idr/attributes.py

```diff
@@ -170,16 +174,217 @@ def get_phenotypes_for_gene(gene_name, screenid=None):
+    screen_to_phenotype_dictionary = {}
+    for scid in genes_scid_list:
+        if scid in organism_screen_idlist:
+            content = [x for x in set(list(totalscreenids))
+                       if x.startswith(scid)]
+            for idx, item in enumerate(content):
+                idx1 = item.index('_')
+                content[idx] = item[idx1+1:]
+            screen_to_phenotype_dictionary[scid] = content
+
+    return [query_genes_dataframe, screen_to_phenotype_dictionary]
+
+
+def get_similar_genes(conn, query_genes_list, screen_to_phenotype_dictionary):
+
+    """
+    Return a multi-dimensional dictionary with the
+    following mapping,
+    similar_genes[screenid][phenotypename] = similar_genes_list
+    (uses python blitz gateway)
+    """
+
+    similar_genes = {}
+    overlap_genes = {}
+    scid_list = set(list(screen_to_phenotype_dictionary.keys()))
+    for i, sid in enumerate(set(scid_list)):
+
+        phlist = screen_to_phenotype_dictionary[sid]
+        similar_genes[str(sid)] = {}
+        overlap_genes[str(sid)] = {}
+        for phenotype in np.unique(phlist):
+
+            args = {
+                "name": "Phenotype Term Name",
+                "value": phenotype,
+                "ns": "openmicroscopy.org/mapr/phenotype",
+                "ns2": "openmicroscopy.org/mapr/gene",
+                "s_id": sid
+            }
+
+            cc = attributes_by_attributes(conn, **args)
+            dataframe = pandas.DataFrame.from_dict(cc)
+
+            if dataframe.empty:
+                continue
+
+            gene_list = []
+            for x in dataframe.iloc[:, 0]:
+
+                key = x[0]
+                value = x[1]
+
+                if key == "Gene Identifier" and value.startswith("EN"):
+                    id = value
+                    gene_list.append(id)
+                if key == "Gene Symbol":
+                    genesym = value
+                    gene_list.append(genesym)
+
+            ov_genes = set(gene_list).intersection(query_genes_list)
+
+            remove_duplicates = []
+            removed_genes = []
+            for g in ov_genes:
+                converted = g
+                if g.startswith('ENSG'):
+                    converted = ensembleid_to_genesymbol(g)
+                    removed_genes.append(g)
+                remove_duplicates.append(converted)
+
+            ov_genes = set(remove_duplicates)
+
+            if len(ov_genes) != 0:
+                setdiff_genes = set(gene_list) - ov_genes - set(removed_genes)
+                similar_genes[str(sid)][phenotype] = list(setdiff_genes)
+                overlap_genes[str(sid)][phenotype] = list(ov_genes)
+
+        progress(i+1, len(set(scid_list)), status='Iterating through screens')
+
+    return [similar_genes, overlap_genes]
+
+
+def get_organism_screenids(idr_base_url, session, organism):
+
+    """
+    Return a list of screen ids in IDR
+    for given case insensitive organism. (Uses the json api)
```

Comment:
This is part of the public API documentation for the library, is there a reason to mention the implementation details `json api`?

Reply:
since the module contains both blitz gateway and json api implementations for the same process, I have mentioned them in the file. Happy to remove them if its not useful.

Reply:
We likely need to review what's using `idr.connections` and what's using `requests`. It might be that they should be more fully separated. *OR* we intend to move everything to JSON and therefore `connections` is just a temporary workaround.